### PR TITLE
Update MainchainCrossChainUpdate command execute function based on LIP0053 update

### LINKED ISSUE #8017: Update `MainchainCrossChainUpdate` command execute function based on LIP0053 update
### Description

- Update `execute` function of `MainchainCrossChainUpdate` command based on updated specification [LIP0053:MainchainCrossChainUpdate:execution](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#execution)
- Add new test cases based on the update

### Acceptance Criteria

- Should have all the unit tests passing

### Additional Information

- Related LIP Update PR: https://github.com/LiskHQ/lips/pull/187
- Related PR: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md
- Duplicate issue https://github.com/LiskHQ/lisk-sdk/issues/7982

### PR DESCRIPTION
**Pull request by sitetester**
### What was the problem?

This PR resolves #8017

### How was it solved?
Mainly base_cross_chain_update_command.ts::executeCommon part is updated

### CODE REVIEW

File: framework/src/modules/interoperability/types.ts [hunk 1]
```diff
@@ -42,6 +44,17 @@ export interface CCMsg {
...
+	crossChainCommand: '',
+	fee: BigInt(0),
+	module: '',
+	nonce: BigInt(0),
+	params: EMPTY_BYTES,
+	receivingChainID: EMPTY_BYTES,
+	sendingChainID: EMPTY_BYTES,
+	status: 0,
+};
+
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/types.ts:57]: - Add LIP reference https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#empty-cross-chain-message
- Order the properties as per `ccmSchema` and the [LIP: EMPTY_CCM](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#empty-cross-chain-message)

File: framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts [hunk 2]
```diff
@@ -451,7 +445,8 @@ describe('BaseCrossChainUpdateCommand', () => {
 			expect(command['_tokenMethod'].initializeUserAccount).not.toHaveBeenCalled();
 		});
 
-		it('should terminate the chain and add an event when ccm format is invalid', async () => {
+		// decoding
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts:448]: I think we can remove this comment, either we can encapsulate tests within describe with topic or `it('...')` description are good enough?

File: framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts [hunk 3]
```diff
@@ -486,19 +480,19 @@ describe('BaseCrossChainUpdateCommand', () => {
...
+					ccm: EmptyCCM,
 					result: CCMProcessedResult.DISCARDED,
+					code: CCMProcessedCode.INVALID_CCM_DECODING_EXCEPTION,
 				},
 			);
 		});
 
-		// after `validateFormat`, we should check `!context.chainID.equals(ccm.receivingChainID)`
-		it('should terminate the chain and add an event when CCM is not directed to the sidechain', async () => {
+		// format
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts:494]: same here

File: framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts [hunk 4]
```diff
@@ -728,84 +763,62 @@ describe('BaseCrossChainUpdateCommand', () => {
...
 		});
 
-		it('should update certificate when certificate is not empty', async () => {
-			await expect(command['executeCommon'](executeContext, true)).resolves.toEqual([
-				expect.toBeArrayOfSize(params.inboxUpdate.crossChainMessages.length),
-				true,
-			]);
-			expect(internalMethod.updateCertificate).toHaveBeenCalledTimes(1);
-		});
+		it('should update validators if activeValidatorsUpdate is not empty', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts:806]: I think there should be separate test cases to cover, 
`if len(ccu.params.certificate) > 0 updateCertificate(ccu)`
`if ccu.params.inboxUpdate is not empty and len(ccu.params.certificate) > 0: updatePartnerChainOutboxRoot(ccu)`

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: framework/src/modules/interoperability/base_cross_chain_update_command.ts [hunk 5]
```diff
@@ -103,68 +105,108 @@ export abstract class BaseCrossChainUpdateCommand<
...
+
+		const { blsKeysUpdate, bftWeightsUpdate, bftWeightsUpdateBitmap } =
+			params.activeValidatorsUpdate;
+
 		if (
-			!emptyActiveValidatorsUpdate(params.activeValidatorsUpdate) ||
-			params.certificateThreshold !== sendingChainValidators.certificateThreshold
+			blsKeysUpdate.length > 0 ||
+			bftWeightsUpdate.length > 0 ||
+			bftWeightsUpdateBitmap !== EMPTY_BYTES ||
```
Comment (shuse2) [inline framework/src/modules/interoperability/base_cross_chain_update_command.ts:197]: original condition is correct, and buffer cannot be compared with `!==`

File: framework/src/modules/interoperability/base_cross_chain_update_command.ts [hunk 6]
```diff
@@ -103,68 +105,108 @@ export abstract class BaseCrossChainUpdateCommand<
...
+	}
+
+	protected async afterExecuteCommon(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+	) {
+		const { params } = context;
+
+		// Update sidechain validators.
+		const chainValidatorsStore = this.stores.get(ChainValidatorsStore);
+		const validators = await chainValidatorsStore.get(context, params.sendingChainID);
```
Comment (shuse2) [inline framework/src/modules/interoperability/base_cross_chain_update_command.ts:189]: Minor point, but no need to split the declaration here?

File: framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts [hunk 7]
```diff
@@ -510,43 +504,49 @@ describe('BaseCrossChainUpdateCommand', () => {
...
+					ccm: { ...EmptyCCM, params: EMPTY_BYTES },
 					result: CCMProcessedResult.DISCARDED,
+					code: CCMProcessedCode.INVALID_CCM_VALIDATION_EXCEPTION,
 				},
 			);
 		});
 
-		// next, we should check `!ccm.sendingChainID.equals(params.sendingChainID)`
-		// params.sendingChainID is defaultSendingChainID (Buffer.from([0, 0, 2, 0])
+		// The CCM must come from the sending chain.
```
Comment (shuse2) [inline framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts:534]: same as above @ishantiw 's comment

File: framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts [hunk 8]
```diff
@@ -634,18 +621,79 @@ describe('BaseCrossChainUpdateCommand', () => {
...
+				ccm.receivingChainID,
+				{
+					ccm,
+					result: CCMProcessedResult.DISCARDED,
+					code: CCMProcessedCode.INVALID_CCM_ROUTING_EXCEPTION,
+				},
+			);
+		});
+
+		// The CCM must come be directed to the sidechain, unless it was bounced on the mainchain.
```
Comment (shuse2) [inline framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts:635]: same as above @ishantiw 's comment

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/interoperability/base_cross_chain_update_command.ts [hunk 9]
```diff
@@ -295,9 +328,8 @@ export abstract class BaseCrossChainUpdateCommand<
 			});
 		}
 	}
-	/**
-	 * @see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#bounce
-	 */
+
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#bounce
```
Comment (ishantiw) [inline framework/src/modules/interoperability/base_cross_chain_update_command.ts:332]: why we changed this to single comment? I thought we are following TypeDoc way to mention LIP references?

File: framework/src/modules/interoperability/constants.ts [hunk 10]
```diff
@@ -85,3 +85,15 @@ export const EVENT_NAME_CCM_PROCESSED = 'ccmProcessed';
 export const EVENT_NAME_CCM_SEND_SUCCESS = 'ccmSendSucess';
 
 export const CONTEXT_STORE_KEY_CCM_PROCESSING = 'CONTEXT_STORE_KEY_CCM_PROCESSING';
+
+// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#empty-cross-chain-message
```
Comment (ishantiw) [inline framework/src/modules/interoperability/constants.ts:89]: same here

**Review by ishantiw — APPROVED** [review]
Looks good, couple of minor comments related to TypeDoc but we can keep in mind for future PRs

**Pull request merged**